{
  try {
    Task task=createTask(InvokableWithExceptionInInvoke.class);
    task.registerExecutionListener(listener);
    task.run();
    assertEquals(ExecutionState.FAILED,task.getExecutionState());
    assertTrue(task.isCanceledOrFailed());
    assertTrue(task.getFailureCause().getMessage().contains("test"));
    validateTaskManagerStateChange(ExecutionState.RUNNING,task,false);
    validateUnregisterTask(task.getExecutionId());
    validateListenerMessage(ExecutionState.RUNNING,task,false);
    validateListenerMessage(ExecutionState.FAILED,task,true);
  }
 catch (  Exception e) {
    e.printStackTrace();
    fail(e.getMessage());
  }
}
